src/components/Search/Homepage/HomepageSearchAutocomplete.tsx

```diff
@@ -118,7 +128,8 @@ const HomepageSearchAutocomplete: React.FC<{
           onChange={onSelect}
           getOptionSelected={getOptionSelected}
           filterOptions={createFilterOptions({
-            matchFrom: checkForZipcodeMatch ? 'any' : 'start',
+            matchFrom:
+              checkForZipcodeMatch || checkForStateMatch ? 'any' : 'start',
```

Comment:
if we set `matchFrom` to 'any' whenever the input is 2 letters, it starts checking for matches against any part of a location name, which isnt exactly what we want. for example, an input of 'ct' returns 'Distri**ct** of Columbia' and 'Grand Jun**ct**ion metro, CO':

<img width="730" alt="Screen Shot 2021-02-23 at 4 29 34 PM" src="https://user-images.githubusercontent.com/44076375/108915272-a24a8a80-75fa-11eb-95bb-17a857810651.png">